src/openassetio-python/tests/cmodule/gil/test_managerfactory_gil.py

```diff
@@ -70,6 +71,50 @@ def test_instantiate(
         a_threaded_manager_impl_factory.instantiate("")
 
 
+class Test_HybridPluginSystemManagerImplementationFactory_gil:
+    """
+    Check all methods release the GIL during C++ function body
+    execution.
+
+    See docstring for similar test under `gil/Test_ManagerInterface.py`
+    for details on how these tests are structured.
+    """
+
+    def test_all_methods_covered(self, find_unimplemented_test_cases):
+        """
+        Ensure this test class covers all methods.
+        """
+        unimplemented = find_unimplemented_test_cases(
+            HybridPluginSystemManagerImplementationFactory, self
+        )
+
+        if unimplemented:
+            print("\nSome test cases not implemented. Method templates can be found below:\n")
+            for method in unimplemented:
+                print(
+                    f"""
+    def test_{method}(self, a_threaded_hybrid_impl_factory, mock_manager_impl_factory):
+        a_threaded_hybrid_impl_factory.{method}()
+"""
+                )
+
+        assert unimplemented == []
+
+    def test_identifiers(self, a_threaded_hybrid_impl_factory, mock_manager_impl_factory):
+        mock_manager_impl_factory.mock.identifiers.return_value = []
+        a_threaded_hybrid_impl_factory.identifiers()
```

Comment:
Ah yes, the GIL tests.  A bit of a tricky one to follow.  

In short, we will get an exception if the GIL is not released by the Python bindings when calling `identifiers()`.

We must release the GIL for two reasons
* Performance, in case of long-running methods.
* Deadlock prevention, in case the C++ method implementation wants to execute some Python code.

The test class docstring links to `gil/Test_ManagerInterface.py` where there is a much longer docstring explaining what's going on in these tests.

They require bespoke Python bindings to 
* Assert the GIL is not held when a given method is called
* Execute the method in a separate thread.  This is not strictly necessary, the GIL check is technically enough, but it provides extra confidence.

The code for those Python bindings can be a bit hard to read. We hack some macros from a mocking framework, and it gets pretty ugly.

Reply:
Ah right, required some under the hood knowledge. Thanks.